accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -92,6 +97,18 @@ The internal representation of types is extended with a type `T*` for every type
 `T` to represent legacy pre-NNBD types.  This is discussed further in the legacy
 library section below.
 
+### Future flattening
+
+The **flatten** function is modified as follows:
+
+**flatten**(`T`) is defined by cases on `T`:
+  - if `T` is `S?` then **flatten**(`T`) = **flatten**(`S`)`?`
+  - otherwise if `T` is `S*` then **flatten**(`T`) = **flatten**(`S`)`*`
+  - otherwise if `T` is `FutureOr<S>` then **flatten**(`T`) = `S`
+  - otherwise if `T <: Future` then let `S` be a type such that `T <: Future<S>`
+and for all `R`, if `T <: Future<R>` then `S <: R`; then **flatten**('T') = `S`
+  - otherwise **flatten**('T') = `T`
+
 ### Static errors
 
 We say that a type `T` is **nullable** if `Null <: T` and not `T <: Object`.
```

Comment:
(Currently "not `T <: Object`" is sufficient, right? I guess it's a good idea to have both).

Reply:
No.  If `T` is `X extends Object?` we have "not `T <: Object`", but we don't want to treat `X` as **nullable** - only as **potentially nullable**.